src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -384,12 +384,13 @@ private bool ConnectionNameMatchesCurrentApp(DurableClient client)
         async Task IDurableOrchestrationClient.TerminateAsync(string instanceId, string reason)
         {
             OrchestrationState state = await this.GetOrchestrationInstanceStateAsync(instanceId);
-            if (IsOrchestrationRunning(state))
+            if (IsOrchestrationAvailable(state))
             {
                 // Terminate events are not supposed to target any particular execution ID.
                 // We need to clear it to avoid sending messages to an expired ContinueAsNew instance.
                 state.OrchestrationInstance.ExecutionId = null;
 
+                // NOTE: this taps into the Durable Task repo
```

Comment:
I think we can remove this, since we, as DF devs, are aware of that :)

Reply:
Done